The LNA improves the rx path. Within a simple test setup
it improved the signal from -60dbm to -40dbm.

Fix low tx power on TpLink cpe510 devices

Signed-off-by: Alexander Couzens <lynxis@fe80.eu>

--- a/openwrt/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c
+++ b/openwrt/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c
@@ -30,6 +30,9 @@
 #define CPE510_GPIO_LED_L3     15
 #define CPE510_GPIO_LED_L4     16

+#define CPE510_GPIO_EXTERNAL_LNA0      18
+#define CPE510_GPIO_EXTERNAL_LNA1      19
+
 #define CPE510_GPIO_BTN_RESET  4

 #define CPE510_KEYS_POLL_INTERVAL      20 /* msecs */
@@ -93,6 +96,9 @@ static void __init cpe510_setup(void)
                                        ARRAY_SIZE(cpe510_gpio_keys),
                                        cpe510_gpio_keys);

+       ath79_wmac_set_ext_lna_gpio(0, CPE510_GPIO_EXTERNAL_LNA0);
+       ath79_wmac_set_ext_lna_gpio(1, CPE510_GPIO_EXTERNAL_LNA1);
+
        ath79_register_m25p80(NULL);

        ath79_register_mdio(1, 0);

